### PR TITLE
Powermax resiliency changes

### PR DESCRIPTION
**Pull request by delldubey**
#### Is this a new chart?

No

#### What this PR does / why we need it:
- Podmon installation with the driver
#### Which issue(s) is this PR associated with:

- https://github.com/dell/csm/issues/1082

#### Checklist:

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [ ] Chart Version bumped
- [ ] Variables are documented in the chart README.md
- [ ] Title of the PR starts with the chart name (e.g. `[charts_dir/mychartname]`) if applicable

### CODE REVIEW

**Review by boyamurthy — APPROVED** [review]
LGTM.

**Review by AkshaySainiDell — CHANGES_REQUESTED** [review]

File: charts/csi-powermax/templates/controller.yaml [hunk 1]
```diff
@@ -169,6 +181,33 @@ spec:
                 - {{ .Release.Name }}-controller
             topologyKey: kubernetes.io/hostname
       containers:
+        {{- if hasKey .Values "podmon" }}
+        {{- if eq .Values.podmon.enabled true }}
+        - name: podmon
+          image: {{ required "Must provide the podmon container image." .Values.images.podmon }}
+          imagePullPolicy: {{ .Values.imagePullPolicy }}
+          args:
+                    {{- toYaml .Values.podmon.controller.args | nindent 12 }}
```
Comment (AkshaySainiDell) [inline charts/csi-powermax/templates/controller.yaml:190]: The indentation does not look correct. Please check.

File: charts/csi-powermax/templates/node.yaml [hunk 2]
```diff
@@ -218,12 +245,21 @@ spec:
                   name: {{ .Values.vSphere.vCenterCredSecret }}
                   key: password
             {{- end }}
+            - name: X_CSI_PODMON_ENABLED
+              value: "{{ .Values.podmon.enabled }}"
+            - name: X_CSI_PODMON_ARRAY_CONNECTIVITY_POLL_RATE
+              value: "60"
```
Comment (AkshaySainiDell) [inline charts/csi-powermax/templates/node.yaml:251]: I think this should be read from values.yaml files. Please check.

**Review by AkshaySainiDell — CHANGES_REQUESTED** [review]

**Review by boyamurthy — COMMENTED** [review]
Please add api Port and podmon image in values.yaml file .

**Review by boyamurthy — APPROVED** [review]

**Review by AkshaySainiDell — APPROVED** [review]
LGTM

**Review by chimanjain — APPROVED** [review]

**Pull request merged**